### ISSUE TITLE
Please downgrade System.Collections.Immutable from 9.0.0 to 8.0.0

### ISSUE THREAD

**Issue by FlaggAC** [corvus-dotnet/Corvus.JsonSchema#512]
When running from a source generator, Any reference that is also a dependency of `Microsoft.CodeAnalysis.CSharp` must not be a greater version than referenced by the Microsoft.CodeAnalysis.CSharp project. There is no workaround that I could find, other than to do a complete fork of the library to remove the reference.

`CS1705 Assembly 'Corvus.Json.ExtendedTypes' with identity 'Corvus.Json.ExtendedTypes, Version=4.1.0.0, Culture=neutral, PublicKeyToken=null' uses 'System.Collections.Immutable, Version=9.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version than referenced assembly 'System.Collections.Immutable' with identity 'System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'`

The references in that package are maintained by the visual studio team at Microsoft, separate from the .NET team, and tend to be delayed. We have to wait for the next major version of that to come out, and they don't give out release dates from what I understand. 

The Corvus source generator library sadly isn't sufficient for my needs so I require a dependency on Corvus.Json.ExtendedTypes. I was finally able to get it working when building locally without this dependency (there's nothing in it that requires System.Collections.Immutable version 9.0.0 over version 8.0.0 that I could find).

It would be great if there was at least an ExtendedTypes package maintained that is compatible with the latest version of Microsoft.CodeAnalysis.CSharp at any time. 

BTW: To deal with the .NET Standard "Invalid Program" issue which occurs when running from `dotnet build` (but doesn't occur from Visual Studio), I actually got it to work by having the generator target both .net standard and .net 9, and mute the warning that tells you not to do that. Works like a charm so far, though it's a bit of a hack, but one less hurdle out of the way to get this pipeline into production!

```
  <ItemGroup Condition="'$(BuildingInsideVisualStudio)' == 'true'">
    <ProjectReference Include="..\ACRealms.Roslyn.RealmProps\ACRealms.Roslyn.RealmProps.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" SetTargetFramework="TargetFramework=netstandard2.0" />   
  </ItemGroup>
  <ItemGroup Condition="'$(BuildingInsideVisualStudio)' != 'true'">
    <ProjectReference Include="..\ACRealms.Roslyn.RealmProps\ACRealms.Roslyn.RealmProps.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" SetTargetFramework="TargetFramework=net9.0" />
  </ItemGroup>  
  ```

**Comment by mwadams** [comment]
I have pushed up a new preview build which includes a new package `Corvus.Json.SourceGeneratorTools`

https://www.nuget.org/packages/Corvus.Json.SourceGeneratorTools/4.1.1-preview.2

This is a self contained assembly with a pure netstandard2.0 package and 8.x dependencies which specifically helps you build analysers/source generators. It embeds an analyser version of Corvus.Json.ExtendedTypes and all of the code generation scaffolding in a single package, amongst other things.

I am just working on some sample code which illustrates how to use it and I will provide a link here when that is done (probably over the weekend.)

**Comment by mwadams** [comment]
However, prior to building this library, I am happily running the v9.x System.Text.Json in a SourceGenerator targeting netstandard2.0 - it doesn't need anything net9.0 to do so. So maybe there is another issue?

**Comment by mwadams** [comment]
Ah! Of course.

**Comment by mwadams** [comment]
Closed with V4.1.1
